File: BE/table-service/src/main/java/com/vietnl/tableservice/application/dto/ReservationRequest.java
package com.vietnl.tableservice.application.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class ReservationRequest {
    private UUID tableId;
    private String customerName;
    private String customerPhone;
    private Integer partySize;
    private LocalDateTime reservedAt;
}

File: FE/mobile/app/src/main/java/com/food/order/data/request/ReservationRequest.kt
package com.food.order.data.request

import com.google.gson.annotations.SerializedName

// Khớp field-for-field với ReservationRequest.java bên table-service (application/dto).
data class ReservationCreateRequest(
    @SerializedName("tableId")       val tableId: String,
    @SerializedName("customerName")  val customerName: String,
    @SerializedName("customerPhone") val customerPhone: String?,
    @SerializedName("partySize")     val partySize: Int,
    // ISO local date-time không kèm timezone, vd "2026-08-28T19:30:00" — khớp @DateTimeFormat(iso =
    // DATE_TIME) / LocalDateTime bên BE.
    @SerializedName("reservedAt")    val reservedAt: String
)
